<!doctype html>
<html>
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

        <title>Assembling forest tree genomes: are we there yet?</title>

        <link rel="stylesheet" href="css/reveal.css">
        <link rel="stylesheet" href="css/theme/maehler_dark.css">

        <!-- Theme used for syntax highlighting of code -->
        <link rel="stylesheet" href="lib/css/zenburn.css">

        <!-- Font awesome -->
        <link rel="stylesheet" href="lib/css/fontawesome.css">
        <link rel="stylesheet" href="lib/css/solid.css">
        <link rel="stylesheet" href="lib/css/brands.css">

        <!-- Custom modifications that shouldn't go into the theme -->
        <link rel="stylesheet" href="css/customise.css">

        <!-- Printing and PDF exports -->
        <script>
            var link = document.createElement( 'link' );
            link.rel = 'stylesheet';
            link.type = 'text/css';
            link.href = window.location.search.match( /print-pdf/gi ) ? 'css/print/pdf.css' : 'css/print/paper.css';
            document.getElementsByTagName( 'head' )[0].appendChild( link );
        </script>
    </head>
    <body>
        <div class="slidelink"><a href=""></a></div>
        <div class="twitterlink"><a href="https://twitter.com/nmaehler"><i class="fab fa-twitter"></i>@nmaehler</a></div>
        <div class="reveal">
            <div class="slides">
                <section data-background-image="https://farm6.staticflickr.com/5690/21457742918_964178f889_o_d.jpg"
                         data-background-opacity="0.2">
                    <!-- Background photo: https://www.flickr.com/photos/befuddledsenses/21457742918/ -->
                    <h2>Assembly of forest tree genomes: are we there yet?</h2>
                    <p>Niklas Mähler</p>
                    <p>UPSC seminar <time datetime="2020-03-02T09:30:00">2020-03-02</time></p>
                </section>

                <section>
                    <section>
                        <h2>What is genome assembly?</h2>
                    </section>

                    <section>
                        <p>Recreating a genome sequence from short fragments</p>
                        <img src="http://dna.cs.byu.edu/bio465/puzzle.png" alt="Genome assembly is a complex puzzle">
                        <aside class="notes">
                            <p>
                            How these short fragments are obtained has changed over time, and the last 10-15 years there have been tremendous advancements in this area.
                            </p>
                        </aside>
                    </section>

                    <section>
                        <h2>Data</h2>
                        <div class="side-by-side-container">
                            <div style="transform: rotate(270deg);">
                                <p>Time</p>
                            </div>
                            <div>
                                <svg width="12" height="500" viewBox="0 0 10 500" xlmns="http://www.w3.org/2000/svg">
                                    <line x1="5" y1="0" x2="5" y2="500" stroke="white" stroke-width=2 />
                                    <line x1="5" y1="500" x2="10" y2="490" stroke="white" stroke-width=2 />
                                    <line x1="5" y1="500" x2="0" y2="490" stroke="white" stroke-width=2 />
                                </svg>
                            </div>
                            <div class="smaller">
                                <dl>
                                    <div class="fragment"
                                        style="background-color: #B6B007; padding: 1ex 1em;">
                                        <dt>Sanger sequencing</dt>
                                        <dd>long reads, high accuracy, low throughput</dd>
                                    </div>

                                    <div class="fragment"
                                        style="background-color: #B45505; padding: 1ex 1em;">
                                        <dt>454 sequencing</dt>
                                        <dd>medium reads, high accuracy, high throughput</dd>

                                        <dt>Illumina sequencing</dt>
                                        <dd>short reads, high accuracy, high throughput</dd>
                                    </div>

                                    <div class="fragment"
                                        style="background-color: #B30004; padding: 1ex 1em;">
                                        <dt>PacBio SMRT sequencing</dt>
                                        <dd>long reads, lower accuracy, high throughput</dd>

                                        <dt>Nanopore sequencing</dt>
                                        <dd>longer reads, lower accuracy, high throughput</dd>

                                        <dt>PacBio HiFi sequencing</dt>
                                        <dd>long reads, high accuracy, high throughput</dd>
                                    </div>
                                </dl>
                            </div>
                        </div>
                        <aside class="notes">
                            <p>
                            Here I list some of them, in chronological order.
                            </p>
                            <ul>
                                <li>Sanger: Frederick Sanger developed in the 70s, Nobel prize in 1980. First generation sequencing.</li>
                                <li>454 Sequencing: pyrosequencing technology developed at KTH in Stockholm during mid 90s.</li>
                                <li>Illumina sequencing: technology developed in mid 90s &rarr; Solexa &rarr; Illumina. Second, or next generation sequencing.</li>
                                <li>PacBio: Single molecule real-time sequencing, commercialised in 2011. Third generation sequencing.</li>
                            </ul>
                        </aside>
                    </section>

                    <section data-background-color="#FFFFFF">
                        <h3>Short reads vs long reads</h3>
                        <div class="side-by-side-container">
                            <figure style="width: 45%;">
                                <img src="img/horse_puzzle.jpeg" alt="">
                                <figcaption><a href="https://twitter.com/iddux/status/1210915509421694979"><i class="fab fa-twitter"></i> @iddux</a></figcaption>
                            </figure>
                            <figure style="width: 45%;">
                                <img src="img/horse_puzzle_big_pieces.jpeg" alt="">
                                <figcaption><a href="https://twitter.com/olga_francino/status/1210924674856689665"><i class="fab fa-twitter"></i> @olga_francino</a></figcaption>
                            </figure>
                        </div>
                    </section>

                    <section>
                        <figure>
                            <img src="img/de_novo_assembly.svg" alt="Schematic figure of de-novo genome assembly">
                        </figure>
                        <aside class="notes">
                            <p>How this is done in practice I will leave out for now</p>
                        </aside>
                    </section>

                    <section>
                        <h2>N50</h2>
                        <figure>
                            <img src="img/n50.svg" alt="N50 explained">
                        </figure>
                    </section>
                </section>

                <section>
                    <section>
                        <h2>The genomes we are working on</h2>
                    </section>

                    <section>
                        <h2><i>Populus tremula</i> v2</h2>
                        <div class="side-by-side-container">
                            <figure style="width: 30%">
                                <img src="img/campus_aspen.jpg" alt="Populus tremula (European aspen)">
                                <figcaption>Photo: Stefan Jansson</figcaption>
                            </figure>
                            <div>
                                <ul>
                                    <li>European aspen</li>
                                    <li>450 Mbp</li>
                                    <li>Diploid</li>
                                    <li>19 chromosomes</li>
                                    <li>Assembly ready and available at <a href="https://popgenie.org">PopGenie.org</a></li>
                                    <li>Version 1 in 2018 (<a href="https://doi.org/10.1073/pnas.1801437115">Lin <i>et al.</i></a>)</li>
                                </ul>
                            </div>
                        </div>
                    </section>

                    <section>
                        <h2><i>Picea abies</i> v2</h2>
                        <div class="side-by-side-container">
                            <figure style="width: 30%">
                                <img src="img/z4006.jpg" alt="Picea abies (Norway spruce)">
                            </figure>
                            <div>
                                <ul>
                                    <li>Norway spruce</li>
                                    <li>20 Gbp</li>
                                    <li>Diploid</li>
                                    <li>12 chromosomes</li>
                                    <li>Version 1 in 2013 (<a href="https://doi.org/10.1038/nature12211">Nystedt <i>et al.</i></a>)</li>
                                </ul>
                            </div>
                        </div>
                    </section>

                    <section>
                        <h2><i>Pinus sylvestris</i> v1</h2>
                        <div class="side-by-side-container">
                            <figure style="width: 30%;">
                                <img src="img/pinus_sylvestris.jpg" alt="Pinus sylvestris (Scots pine)">
                                <figcaption>Photo: <a href="https://www.flickr.com/photos/74738817@N07/28360519966/in/photolist-Kd85wb-KZhyWM-PJaiJL-N1k8U6-MQF5Cx-MWyYM5-23zZd6r-M33VSr-MVcX8b-24TmJKY-KgNVvW-H1E4DB-Kd87zj-PxnX7Q-21ya6e8-Ys3WdA-LdBn1D-28fNyyJ-24TmKrN-Z29hV6-25z8puS-BV6BbG-26w89Uh-GVRfbB-GVRhDH-XZUj9H-25dffKa-Dyeet9-D3WYfn-CtiB-DS6j2M-DPVbPd-MK6XNM-bkBq7U-LR5c5L-D3BSCQ-ruCxnx-D3BEmA-YuYJKr-XebvPL-4sv2AR-rqngK8-x9S48M-XebwRW-7oGhhC-VX8AbY-myjSaf-myhZwz-myikJr-gfiopM">Andreas Rockstein</a></figcaption>
                            </figure>
                            <ul>
                                <li>Scots pine</li>
                                <li>22 Gbp</li>
                                <li>Diploid</li>
                                <li>12 chromosomes</li>
                            </ul>
                        </div>
                    </section>
                </section>

                <section>
                    <section>
                        <h3>Why reassemble existing genomes?</h3>
                        <dl>
                            <dt class="fragment" data-fragment-index="1">Advances in sequencing</dt>
                            <dd class="fragment" data-fragment-index="1">Makes new things possible/cheaper</dd>

                            <dt class="fragment" data-fragment-index="2">Advances in computation</dt>
                            <dd class="fragment" data-fragment-index="2">Better hardware and new algorithms</dd>

                            <dt class="fragment" data-fragment-index="3">We're curious</dt>
                            <dd class="fragment" data-fragment-index="3">The current genomes just aren't good enough</dd>
                        </dl>
                        <aside class="notes">
                            <ul>
                                <li>Fragmented gene space</li>
                                <li>Missing regions</li>
                                <li>Low contiguity</li>
                            </ul>
                        </aside>
                    </section>

                    <section>
                        <p>
                        Especially conifers are of great importance, both economically and ecologically.
                        </p>
                        <aside class="notes">
                            <ul>
                                <li>Roughly 10% of Swedish export is related to the forest industry.</li>
                                <li>Ability to improve growth and various traits is important.</li>
                            </ul>
                        </aside>
                    </section>

                    <section>
                        <figure class="stretch">
                            <img src="img/breeding_cycle.svg" alt="Spruce breeding cycle">
                        </figure>
                    </section>

                    <section data-background-image="img/gudrun_logs.png"
                             data-background-size="contain"
                             data-background-color="#000000">
                        <aside class="notes">
                            <p>
                            After the storm Gudrun in 2004.
                            This is the old military airstrip in Byholma, Småland.
                            In the picture there is about 1 million m<sup>3</sup> of wood, and this is rougly equivalent to the biomass production in Swedish forests in one day.
                            You can then imagine what just a small increase in for example growth rate could result in.
                            </p>
                        </aside>
                    </section>

                    <section>
                        <h3>Conifer genome projects</h3>
                        <ul>
                            <li>80 million SEK over 6 years</li>
                            <li>Improve growth rate with 30&ndash;40%</li>
                            <li>Improve wood properties</li>
                        </ul>
                        <figure style="margin-top: 1em">
                            <img src="img/kaw_logo_white.svg" alt="Knut & Alice Wallenberg Foundation">
                        </figure>
                    </section>

                    <section class="smaller" data-background-image="img/the_matrix.jpg",
                        data-background-opacity="0.3">
                        <div class="side-by-side-container wrap">
                            <div style="width: 50%;">
                                <h4>Genome</h4>
                                <ul>
                                    <li>50X PacBio WGS</li>
                                    <li>50X Illumina NovaSeq WGS</li>
                                    <li>50X 10X Chromium</li>
                                    <li>BioNano</li>
                                    <li>Hi-C</li>
                                </ul>
                            </div>

                            <div style="width: 50%;">
                                <h4>Transcriptome</h4>
                                <ul>
                                    <li>RNA-Seq</li>
                                    <li>IsoSeq</li>
                                </ul>
                            </div>

                            <div style="width: 50%;">
                                <h4>Resequencing</h4>
                                <ul>
                                    <li>10X WGS, 1,000 trees</li>
                                </ul>
                            </div>

                            <div style="width: 50%;">
                                <h4>SNP-chip</h4>
                                <ul>
                                    <li>800k markers, 10,000 trees</li>
                                </ul>
                            </div>
                        </div>
                        <aside class="notes">
                            <ul>
                                <li>Pacbio: long read sequencing</li>
                                <li>Illumina: short read sequencing</li>
                                <li>10X Chromium: linked reads</li>
                                <li>Hi-C: chromosome conformation capture &rarr; contact maps</li>
                            </ul>
                        </aside>
                    </section>
                </section>

                <section>
                    <section>
                        <h3>Where do we aim?</h3>
                        <aside class="notes">
                            <p>If we ask people what they want from a new assembly, things that are mentioned are things like a phased chromosome level assembly with the majority of repeats resolved.</p>
                            <p>When it comes to conifers, this is essentially what is being asked for:</p>
                        </aside>
                    </section>

                    <section data-background-color="#FFFFFF">
                        <figure class="stretch">
                            <img src="img/unicorn.jpg" alt="Unicorn, hand-colored copper engraving form children's picture book by Friedrich Justin Bertuch, Weimar, 1792">
                            <figcaption>Friedrich Justin Bertuch, Weimar, 1792</figcaption>
                        </figure>
                    </section>

                    <section>
                        <ul>
                            <li>No genes split across scaffolds</li>
                            <li>Longer up/downstream regions around genes</li>
                            <li>More scaffolds with more than one gene</li>
                        </ul>
                    </section>

                    <section data-background-color="#FFFFFF">
                        <h3><i>Populus tremula</i> v2</h3>
                        <figure>
                            <img src="img/acetyl_salicortin_v2_manhattan.png" alt="Acetyl salicortin GWAS">
                        </figure>
                        <aside class="notes">
                            <p>
                            Here we can take an example from aspen since we have two versions that we can compare.
                            Even though it is the same resequencing data that the SNPs are based on, the signal is much stronger.
                            We don't know the exact reason for this, but we have some guesses:
                            </p>
                            <ul>
                                <li>Repeats have been resolved</li>
                                <li>Missing regions are now included</li>
                            </ul>
                        </aside>
                    </section>

                    <section>
                        <h3>Assembled conifers</h3>
                        <table class="tiny">
                            <thead>
                                <tr>
                                    <th>Year</th>
                                    <th>Species</th>
                                    <th>Genome size (Gbp)</th>
                                    <th>Data</th>
                                    <th>Contig N50 (kbp)</th>
                                    <th>Scaffold N50 (kbp)</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr>
                                    <td>2016</td>
                                    <td><i>Pinus lambertania</i></td>
                                    <td>31</td>
                                    <td>Illumina, 10X</td>
                                    <td>4.3</td>
                                    <td>247</td>
                                </tr>
                                <tr>
                                    <td>2017</td>
                                    <td><i>Pseudotsuga menziesii</i></td>
                                    <td>16</td>
                                    <td>Illumina</td>
                                    <td>44</td>
                                    <td>341</td>
                                </tr>
                                <tr>
                                    <td>2017</td>
                                    <td><i>Pinus taeda</i></td>
                                    <td>22</td>
                                    <td>Illumina, PacBio</td>
                                    <td>25</td>
                                    <td>108</td>
                                </tr>
                                <tr>
                                    <td>2018</td>
                                    <td><i>Larix sibirica</i></td>
                                    <td>18</td>
                                    <td>Illumina</td>
                                    <td>1.1</td>
                                    <td>6</td>
                                </tr>
                                <tr>
                                    <td>2019</td>
                                    <td><i>Abies alba</i></td>
                                    <td>18</td>
                                    <td>Illumina</td>
                                    <td>2.5</td>
                                    <td>14</td>
                                </tr>
                                <tr>
                                    <td>Ongoing</td>
                                    <td><i>Sequoia sempervirens</i></td>
                                    <td>27</td>
                                    <td>Illumina, Nanopore</td>
                                    <td>&mdash;</td>
                                    <td>110</td>
                                </tr>
                                <tr>
                                    <td>Ongoing</td>
                                    <td><i>Sequoiadendron giganteum</i></td>
                                    <td>8</td>
                                    <td>Illumina, Nanopore, Hi-C</td>
                                    <td>&mdash;</td>
                                    <td>690,000</td>
                                </tr>
                            </tbody>
                        </table>
                        <aside class="notes">
                            <ul>
                                <li>Sugar pine</li>
                                <li>Douglas fir</li>
                                <li>Loblolly pine</li>
                                <li>Siberian larch</li>
                                <li>Silver fir</li>
                                <li>Coast redwood</li>
                                <li>Giant sequoia</li>
                            </ul>
                        </aside>
                    </section>
                </section>

                <section>
                    <section>
                        <h3>Where are we now?</h3>
                    </section>

                    <section>
                        <figure class="stretch">
                            <img src="img/data_flowchart.svg" alt="Assembly workflow flowchart">
                        </figure>
                    </section>

                    <section>
                        <table class="small">
                            <thead>
                                <tr>
                                    <th>Assembly</th>
                                    <th>Number of scaffolds</th>
                                    <th>Max size</th>
                                    <th>Assembly size (Gbp)</th>
                                    <th>N50</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr class="fragment">
                                    <td>Spruce v1</td>
                                    <td>10,253,694</td>
                                    <td>208,095</td>
                                    <td>12.3</td>
                                    <td>5,206</td>
                                </tr>
                                <tr class="fragment">
                                    <td>Spruce v2</td>
                                    <td>312,866</td>
                                    <td>2,145,498</td>
                                    <td>19.1</td>
                                    <td>139,737</td>
                                </tr>
                            </tbody>
                        </table>
                    </section>

                    <section data-transition="slide-in none">
                        <figure class="stretch">
                            <img src="img/n50_1.png" alt="Read and contig N50 of Spruce v1">
                        </figure>
                    </section>

                    <section data-transition="none slide-out">
                        <figure class="stretch">
                            <img src="img/n50_2.png" alt="Read and contig N50 of Spruce v2 (so far)">
                        </figure>
                    </section>

                    <section>
                        <figure class="stretch">
                            <img src="img/illumina_global_coverage.png" alt="Illumina sequence coverage on the spruce contigs">
                        </figure>

                        <aside class="notes">
                            <ul>
                                <li>One peak at about 50X coverage, which is to be expected (since this was the sequencing depth).</li>
                                <li>Other peak at half coverage, which would be haplotypes that have not been collapsed.</li>
                            </ul>
                            <p>
                            More than 10% of the genome without any coverage at all. Still, 99.5% of the reads align to the assembly, so what are these regions without any coverage?
                            </p>
                        </aside>
                    </section>

                    <section>
                        <figure class="stretch">
                            <img src="img/coverage_vs_gc.png" alt="Coverage vs GC content">
                        </figure>
                    </section>
                </section>

                <section>
                    <section>
                        <h2>Challenges</h2>
                        <div class="side-by-side-container" style="justify-content: space-around;">
                            <div>
                                <h3>Technical</h3>
                                <ul>
                                    <li>Amount of data</li>
                                    <li>Computation time</li>
                                </ul>
                            </div>
                            <div>
                                <h3>Biological</h3>
                                <ul>
                                    <li>Repeats</li>
                                    <li>Long introns</li>
                                </ul>
                            </div>
                        </div>
                        <aside class="notes">
                            <h3>Data</h3>
                            <p>
                            The FASTA file for the sequencing data for pine is about 2 TB in size.
                            If this would have been a song on spotify, streamed at high quality,
                            the song would be 28,500 hours long, or about 3 years and 3 months.
                            </p>

                            <h3>Computation time</h3>
                            <p>
                            So far, we have used over 3 million CPU hours on UPPMAX.
                            If this would have been run on a computer with a single CPU, it
                            would correspond to about 340 years of computation.
                            </p>
                        </aside>
                    </section>
                </section>

                <section data-background-image="https://farm5.staticflickr.com/4697/26131146888_198c5a06ce_k.jpg"
                         data-background-opacity="0.4">
                    <!-- Background photo: https://www.flickr.com/photos/inkandelin/26131146888 -->
                    <section>
                        <h2>So, are we there yet?</h2>
                    </section>

                    <section>
                        <figure class="stretch">
                            <img src="img/data_flowchart.svg" alt="Assembly workflow flowchart">
                        </figure>
                        <aside class="notes">
                            <p>
                            In some aspects we still have a long way to go.
                            Things that are basic in other genome projects does not seem to be basic for us; a lot of trail and error.
                            However, if we look at were we are at the momemnt, we've gotten quite a way towards a final assembly.
                            </p>
                        </aside>
                    </section>

                    <section>
                        <div class="side-by-side-container">
                            <ul>
                                <li class="fragment" data-fragment-index="1">Assembly size</li>
                                <li class="fragment" data-fragment-index="2">N50</li>
                                <li class="fragment" data-fragment-index="3">100 kbp contigs</li>
                            </ul>
                            <ul style="list-style: none;">
                                <li class="fragment" data-fragment-index="1">12 Gbp &rarr; 19 Gbp</li>
                                <li class="fragment" data-fragment-index="2">0.6 kbp &rarr; 140 kbp</li>
                                <li class="fragment" data-fragment-index="3">21 Mbp &rarr; 12 Gbp</li>
                            </ul>
                        </div>
                    </section>
                </section>

                <section data-background-image="https://farm6.staticflickr.com/5690/21457742918_964178f889_o_d.jpg"
                         data-background-opacity="0.2">
                    <!-- Background photo: https://www.flickr.com/photos/befuddledsenses/21457742918/ -->
                    <h3>Acknowledgements</h3>
                    <div style="display: flex; flex-wrap: wrap; justify-content: space-between; align-content: space-between; align-items: flex-start; font-size: 0.6em;">
                        <div class="acknowledgement smaller">
                            <h4>Umeå</h4>
                            <ul>
                                <li>Nathaniel Street</li>
                                <li>Ove Nilsson</li>
                                <li>Harry Wu</li>
                                <li>Nicolas Delhomme</li>
                                <li>Bastian Schiffthaler</li>
                                <li>Kathryn Robinson</li>
                                <li>Alexis Sullivan</li>
                            </ul>
                        </div>

                        <div class="acknowledgement smaller">
                            <figure>
                                <img style="width: 10em;" src="img/scilifelab_green_white.png" alt="SciLifeLab">
                            </figure>
                            <ul>
                                <li>Tomas Larsson</li>
                                <li>Per Unneberg</li>
                                <li>Manfred Grabherr</li>
                                <li>Ignas Bunikis</li>
                                <li>Remi-André Olsen</li>
                                <li>Carl-Johan Rubin</li>
                                <li>Ellen Sherwood</li>
                                <li>Ulf Gyllensten</li>
                                <li>Björn Nystedt</li>
                            </ul>
                        </div>

                        <div class="acknowledgement">
                            <figure>
                                <img src="img/umu-logo-neg-EN.png" alt="Umeå University">
                            </figure>
                            <figure style="width: 30%; margin-top: 1em;">
                                <img src="img/upsc_logo.png" alt="UPSC">
                            </figure>
                        </div>

                        <div class="acknowledgement smaller" style="width: 30%">
                            <figure style="width: 50%; margin-bottom: -1em;">
                                <img src="img/slu_logo_vit.png" alt="SLU">
                            </figure>
                            <ul>
                                <li>Pär Ingvarsson</li>
                            </ul>
                        </div>

                        <div class="acknowledgement smaller" style="width: 33%;">
                            <h4>Computing</h4>
                            <figure style="width: 50%; margin-bottom: 1em;">
                                <img src="img/snic_logo.svg" alt="SNIC logotype">
                            </figure>
                            <ul>
                                <li>UPPMAX</li>
                            </ul>
                        </div>

                        <div class="acknowledgement smaller" style="width: 33%">
                            <h4>Funding</h4>
                            <figure style="margin-top: 1em;">
                                <img style="width: 10em;" src="img/kaw_logo_white.svg" alt="KAW">
                            </figure>
                        </div>
                    </div>
                </section>
            </div>
        </div>

        <script src="js/reveal.js"></script>

        <script>
            // More info about config & dependencies:
            // - https://github.com/hakimel/reveal.js#configuration
            // - https://github.com/hakimel/reveal.js#dependencies
            Reveal.initialize({
                hash: true,
                navigationMode: 'linear',
                dependencies: [
                    { src: 'plugin/markdown/marked.js' },
                    { src: 'plugin/markdown/markdown.js' },
                    { src: 'plugin/notes/notes.js', async: true },
                    { src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } }
                ],
                history: true
            });
        </script>
    </body>
</html>
